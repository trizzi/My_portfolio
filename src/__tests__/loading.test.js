import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Loading from '../components/loading/Loading';
import { ThemeContextProvider } from '../context/ThemeContext';

test('renders loading component with correct styles', () => {
	// Render the Loading component wrapped in ThemeContextProvider
	const { container } = render(
		<ThemeContextProvider>
			<Loading />
		</ThemeContextProvider>,
	);

	// Check if the loading container is rendered
	const loadingContainer = document.querySelector('.container');
	expect(loadingContainer).toBeInTheDocument();

	// Check if the loading spinner is rendered
	const loadingSpinner = document.querySelector('.animate-spin');
	expect(loadingSpinner).toBeInTheDocument();

	expect(container).toMatchSnapshot();
});
