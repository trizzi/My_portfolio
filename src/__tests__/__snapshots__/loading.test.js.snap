// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading component with correct styles 1`] = `
<div>
  <div
    class="container mx-auto flex items-center justify-center overflow-hidden h-screen"
  >
    <div
      class="h-40 w-40 rounded-full relative flex items-center justify-center"
      style="background-color: rgb(153, 50, 204);"
    >
      <div
        class="h-36 w-36 bg-white relative rounded-full animate-spin"
      >
        <div
          class="h-36 w-60 bg-white -mt-10 -ml-10"
        />
      </div>
    </div>
  </div>
</div>
`;
